src/components/Header.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import Logo from '../assets/images/Logo.png'
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSearch, faBars } from '@fortawesome/free-solid-svg-icons';
+
+import '../assets/styles/components/Header.css';
+
+// import { mockStudents } from './data';
+
+// const currentStudent = mockStudents[0];
+
+class Header extends React.Component{
+
+    state = {
+        isAuth: false,
+        searchInput: '',
+        isSearchCollapsed: true,
+        isMenuCollapsed: true
+    }
+
+    render(){
+        const handleChange = e => {
```

Comment:
If you want to use class components is better to declare functions outside application render. For example: 

```
import React from "react";
 
class ClassComponent extends React.Component{
    constructor(){
        super();
        this.state={
            count :0
        };
        this.increase=this.increase.bind(this);
    }
     
   increase(){
       this.setState({count : this.state.count +1});
   }
 
    render(){
        return (
            <div style={{margin:'50px'}}>
               <h1>Welcome to Geeks for Geeks </h1>
               <h3>Counter App using Class Component : </h3>
               <h2> {this.state.count}</h2> 
               <button onClick={this.increase}> Add</button>
 
            </div>
        )
    }
}
 
export default ClassComponent;
```

Reply:
Hello @leoneloliveros it's done